html/layout.kk

```diff
@@ -0,0 +1,61 @@
+/*----------------------------------------------------------------------------
+   Copyright 2024, Koka-Community Authors
+
+   Licensed under the MIT License ("The License"). You may not
+   use this file except in compliance with the License. A copy of the License
+   can be found in the LICENSE file at the root of this distribution.
+----------------------------------------------------------------------------*/
+module html/layout
+
+import html/html
+
+pub named effect page
+  fun page-title(): string
+  fun body-content(): component
+  fun body-begin-scripts(): component
+  fun body-end-scripts(): component
+
+pub value struct page-builder<e>
+  page-head : (ev<page>) -> <html-tree|e> ()
+  page-body : (ev<page>) -> <html-tree|e> ()
+
+pub val tmpl/basic : forall<e,s> page-builder<<pure|e>> = Page-builder(
```

Comment:
@TimWhiting what's the `s` here?

Reply:
I started designing a scoped API and then reverted it due to concerns about complexity in the types and some small runtime performance penalties. It probably wouldn't be actually bad in practice, but I wanted some more time to think through the implications, just forgot to remove the scope variable here.